handlers/ExecuteBlockActionHandler.ts

```diff
@@ -150,50 +149,85 @@ export class ExecuteBlockActionHandler {
                 }
                 break;
             }
-            case Modals.SHARE_ACTION: {
-                const persis = this.read.getPersistenceReader();
-                const association_input = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, `${user.id}#share_input`);
-                const share_record = await persis.readByAssociation(association_input);
-                if (share_record) {
-                    let input_str = share_record[0]['share_input'] as string
-                    await sendMessage(
-                            this.modify,
-                            room,
-                            user,
-                            input_str,
-                        );
-                } else {
-                    this.app.getLogger().debug("error: no sharing content");
+            case Modals.SHARE_GITHUB_INPUT_ACTION: {
+                if (value) {
+                    const association = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, `${user.id}#share_github_input`);
+                    await this.persistence.updateByAssociation(association, { share_github_input: value }, true);
                 }
                 break;
             }
-            case Modals.GEN_ACTION: {
-                const persis = this.read.getPersistenceReader();
-                const association_input = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, `${user.id}#gen_input`);
-                const gen_record = await persis.readByAssociation(association_input);
-                if (gen_record) {
-                    let input_str = gen_record[0]['gen_input'] as string
-                    input_str = input_str.substring(0,Math.min(1000, input_str.length))
-                    handler.generateCodeFromParam(input_str);
-                } else {
-                    this.app.getLogger().debug("error: no gen command");
+            case Modals.SHARE_GIST_INPUT_ACTION: {
+                if (value) {
+                    const association = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, `${user.id}#share_gist_input`);
+                    await this.persistence.updateByAssociation(association, { share_gist_input: value }, true);
```

Comment:
`share_gist_input` we need to ensure that the value exists.

Reply:
this is already judged by the if condition